mvp-spec/partials-updater/src/interpretor.ts

```diff
@@ -81,6 +95,34 @@ async function interpretPartialBeginToken(token: LexerToken) : Promise<string> {
     }
 }
 
+async function transformMermaids(config: PartialConfig) {
+    const lineBreak = getLineBreakSymbol(config.files[0]);
+    const cmdAndPathes = config.files.map(buildMermaidCommand);
+    cmdAndPathes.forEach((val) => {
+        child.execSync(val[0]);
+    });
+    const asyncContents = cmdAndPathes.map((val) => {
+        const assetFile = val[1];
+        const assetPath = getAssetPathRelativeToDocuments(assetFile);
+        const assetName = assetFile.substring(0, assetFile.lastIndexOf('.'));
+        const imgRef = `![${assetName}](${assetPath})`;
+        return imgRef;
+    })
+    const contents = await Promise.all(asyncContents);
+    let joined = contents.join(lineBreak);
+    return joined;
+}
+
+function buildMermaidCommand(partialFile): [string, string] {
+    const fileName = partialFile.substring(0, partialFile.lastIndexOf('.'));
+    const destFile = `generated-${fileName}.svg`;
+    const destPath = getAssetPath(destFile);
+    const srcPath = getPartialPath(partialFile);
+    const binPath = getBinPath('mmdc');
+    const cmd = `"${binPath}" -i "${srcPath}" -o "${destPath}"`;
+    return [cmd, destFile]; 
```

Comment:
If two values need to be return, would be much more explicit to return `{cmd, destFile}` rather than a tuple of string.

Reply:
Can you detail your point of view here, please?
Basically, we return a tuple (which is what it is). Do you mean a more formal object? 

I am making a commit with a type created on the fly. Let me know if it is what you are looking for:

```typescript
function buildMermaidCommand(partialFile): { cmd: string; assetFile: string; } {
    // ...
    return { cmd: cmd, assetFile: destFile};
}
```

Reply:
Yes, this is what I mean. I have added comments above to highlight how it would make the code more readable.

Also, there is a shorthand notation that I think is quite neat:
- change const `destFile` to const `assetFile`
- last statement becomes simply `return { cmd, assetFile}`